### ISSUE TITLE
testing pretrained model - depth

### ISSUE THREAD

**Issue by emergencyd** [StanfordVL/taskonomy#15]
![image](https://user-images.githubusercontent.com/16423998/51422806-69a47f80-1bf0-11e9-87bf-9eb912de6f79.png)

I'm using the pretrained model in folder "rgb2depth". I want to reproduce "loss = 0.35". Which data should I use?

I've tried the "depth_zbuffer" test data, but "l1_loss, loss_g, loss_d_real, loss_d_fake" are around "0.6,0.7,0.9,0.7".

I suppose that I used wrong data or wrong loss... should I use "depth_eucliean" data?

Thank you!

**Comment by b0ku1** [comment]
@alexsax could you please comment on which models we used for testing?

**Comment by b0ku1** [comment]
Also, did you use mask to mask out depth values that are "bad"? Basically we extracted loss values from mesh, and since there are holes in mesh, some depth values in ground truth are too high (we masked these values out during training).

**Comment by alexsax** [comment]
@b0ku1 that's a good point. That might explain the relatively high losses. @emergencyd we reported the l1 loss--so that's the one that you should pay attention to.

One contributing factor might be that the models that we released were trained on an internal set of images that were processed a bit differently than the released data. The internal set always has a FoV of 75 degrees, but the released data has a range of 45-75 degrees. The pretrained networks don't work as well on images with a narrow FoV, like those in the release set. You can verify this for yourself on the Taskonomy demo site.

@emergencyd do you notice that the losses are significantly better for large-fov images?

**Comment by emergencyd** [comment]
![image](https://user-images.githubusercontent.com/16423998/51450179-92508480-1d6a-11e9-9da2-e6fc8565eceb.png)
Thank you! I'll try the rgb-large images. But I still guess there is somthing wrong with the images above

**Comment by b0ku1** [comment]
@emergencyd changing to rgb-large won't change the FoV (field of view) problem. That's a discrepancy between the internal and public dataset. Basically we trained and tested on images with fixed FoV (internal), but for more general use for public, the release dataset has varying FoV.

re mask: @alexsax does the released dataset come with mask?

**Comment by alexsax** [comment]
Seconding @b0ku1 above. 

And no need for an explicit mask—just check for pixels where the depth is equal to (or very close to) the max value, 2^16-1 :)

**Comment by emergencyd** [comment]
now I can see "full_plus", "full", "medium", "small" splits information of the whole dataset, but can't find the fov information of each image~ where should I get them~

Also, do I need to drop out the pixels with extremely high values？ （if I understand it right）

**Comment by alexsax** [comment]
> now I can see "full_plus", "full", "medium", "small" splits information of the whole dataset, but can't find the fov information of each image~ where should I get them~
The pose files :)

> Also, do I need to drop out the pixels with extremely high values？ （if I understand it right）
Yes, exactly.

**Comment by emergencyd** [comment]
1) According to supplementary materiel, I use depth_zbuffer rather than depth_euclidean as my target depth mask. 

2) Then I use the "field_of_view_rads" information to pick the images with larger than 1.3 rads. 

3) When calculating the l1-loss, I use the weight mask (0 where the target mask pixels are larger than 2^16-1) during the loss caculation. But still, the l1-loss is about 0.6. I guess there is something wrong with the use of mask. 
@alexsax

**Comment by hanxiaozuo** [comment]
I would like to learn the design of your transfer net. But in your published code, the "TransferNet" part seems to not exist. Could you please publish it? Thank you!

**Comment by emergencyd** [comment]
`img_t = load_raw_image_center_crop(target_name, color=False)`
`img_t[img_t>=2**16-1] = 0`
`img_t[img_t==0]= np.max(img_t)`

`img_t = cfg['target_preprocessing_fn']( img_t, **cfg['target_preprocessing_fn_kwargs'])`
`img_t = img_t[np.newaxis, :]`

`weight_mask = np.ones(np.shape(img_t))`

`predicted, representation, losses = training_runners['sess'].run([m.decoder_output, m.encoder_output, m.losses], feed_dict={m.input_images: img, m.target_images: img_t, m.masks: weight_mask})`

Above is my code for the case "change the target mask(where pixel larger than 2^16-1 -> second max pixel value). Weighted mask for the l1 calculation is all set to 1. This time, the l1 loss is less than 0.1"
